headers/private/kernel/arch/x86/64/cpu.h

```diff
@@ -5,8 +5,9 @@
 #ifndef _KERNEL_ARCH_X86_64_CPU_H
 #define _KERNEL_ARCH_X86_64_CPU_H
 
-
+#ifdef __cplusplus
```

Comment:
Rather than do this everywhere, I would put these guards inside x86 arch_thread_types itself.